docs/sql/sql.md

```diff
@@ -0,0 +1,475 @@
+> It does not impose any index restrictions like the subscription language.
+
+Examples:
+```sql
+-- Which orders include currently stocked items?
+SELECT order.*
+FROM Orders order
+JOIN Inventory inv
+ON order.item_id = inv.item_id
+```
+
+#### WHERE Clause
+
+```ebnf
+predicate
+    = expr
+    | predicate AND predicate
+    | predicate OR  predicate
+    ;
+
+expr
+    = literal
+    | column
+    | expr op expr
+    ;
+
+op
+    = '='
+    | '<'
+    | '>'
+    | '<' '='
+    | '>' '='
+    | '!' '='
+    | '<' '>'
+    ;
+
+literal
+    = INTEGER
+    | FLOAT
+    | STRING
+    | HEX
+    | TRUE
+    | FALSE
+    ;
+```
+
+The query language supports both conjunctions and disjunctions.
+It also supports `!=` comparisons.
+
+> NOTE: Arithmetic expressions will be supported in the future.
+
+Examples:
+```sql
+SELECT * FROM Inventory WHERE item_id = 1;
+```
+
+#### ORDER BY Clause
+
+```ebnf
+ORDER BY column [ ASC | DESC ] { ',' column [ ASC | DESC ] }
+```
+
+`ORDER BY` sorts the result set by one or more of the projected columns.
+Each column involved must be of a type that allows comparison.
+Ascending is the default behavior if not qualified.
+
+Examples:
+```sql
+-- Sort the items in the inventory by order of increasing price
+SELECT * FROM Inventory ORDER BY price;
+
+-- Sort the items in the inventory by order of decreasing price
+SELECT * FROM Inventory ORDER BY price DESC
+
+#### LIMIT Clause
+
+```ebnf
+LIMIT INTEGER
+```
+
+`LIMIT` restricts the number of rows returned.
+
+Examples:
+```sql
+-- What are the 5 highest priced items in my inventory?
+SELECT * FROM Inventory ORDER BY price LIMIT 5;
+```
+
+### INSERT
+
+```ebnf
+INSERT INTO table [ '(' column { ',' column } ')' ] VALUES '(' literal { ',' literal } ')'
+```
+
+Examples:
+```sql
+-- Inserting one row
+INSERT INTO Inventory (item_id, item_name) VALUES (1, 'health1');
+
+-- Inserting two rows
+INSERT INTO Inventory (item_id, item_name) VALUES (1, 'health1'), (2, 'health2');
+```
+
+### DELETE
+
+```ebnf
+DELETE FROM table [ WHERE predicate ]
+```
+
+Deletes all rows from a table.
+If `WHERE` is specified, only the matching rows are deleted.
+
+Examples:
+```sql
+-- Delete all rows
+DELETE FROM Inventory;
+
+-- Delete all rows with a specific item_id
+DELETE FROM Inventory WHERE item_id = 1;
+```
+
+### UPDATE
+
+```ebnf
+UPDATE table SET [ '(' assignment { ',' assignment } ')' ] [ WHERE predicate ]
+```
+
+Updates column values of existing rows in a table.
+The columns are identified by the `assignment` defined as `column '=' expr`.
+The column values are updated for all rows that match the `WHERE` condition.
+The rows are updated after the `WHERE` condition is evaluated for all rows.
+
+Examples:
+```sql
+-- Update the item_name for all rows with a specific item_id
+UPDATE Inventory SET item_name = 'new name' WHERE item_id = 1;
+```
+
+### SET
+
+```ebnf
+SET var ( TO | '=' ) literal
+```
+
+Updates the value of a system variable.
```

Comment:
System variables are stored in the `st_var` table. Should these variables be specified in this document or somewhere else?

Note that we haven't formally stabilized any of the system variables for 1.0. Right now we have `row_limit` and a few others for logging slow queries.

So my questions are:

- Should they be part of the stable api?
    - If so, which ones?
    - If not, does it make sense for `SET var` and `SHOW var` to be part of the stable sql api?

Reply:
Could we mark them as unstable in the documentation? I would be okay to do that as long as we make it clear to users that they are unstable.

I would lump SET and SHOW as part of the unstable API if all of the variables are also unstable.